tldw_Server_API/tests/Embeddings/test_media_embeddings_failure_classification.py

```
@@ -0,0 +1,99 @@
+import pytest
+
+from tldw_Server_API.app.api.v1.endpoints import embeddings_v5_production_enhanced, media_embeddings
+
+
+@pytest.mark.asyncio
+async def test_storage_failure_after_successful_primary_generation_returns_storage_error_and_skips_fallback(monkeypatch):
+    calls: list[tuple[str, str]] = []
+
+    async def fake_create_embeddings_batch_async(*, texts, provider, model_id, metadata):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `fake_create_embeddings_batch_async`

(ANN202)